Feature/dashboard

marcgarnica13 opened the pull request:
### :pushpin: References

-   **Clubhouse** Ref [ch54]

### :memo: Implementation

- Import as landing page: Select OU, Period and Mapping (with list).
- Download JSON button + Import to instance.

### :art: Screenshots
![Screenshot from 2020-03-27 14-45-09](https://user-images.githubusercontent.com/7198996/77762330-a4e2df80-7039-11ea-9706-199148fb932f.png)
![Screenshot from 2020-03-27 14-40-07](https://user-images.githubusercontent.com/7198996/77761937-14a49a80-7039-11ea-80b7-f84edda417ae.png)
![Screenshot from 2020-03-27 14-40-37](https://user-images.githubusercontent.com/7198996/77761955-1a01e500-7039-11ea-90f9-ec0f1f0bfc67.png)
![Screenshot from 2020-03-27 14-40-49](https://user-images.githubusercontent.com/7198996/77761971-1f5f2f80-7039-11ea-892a-a4f71fd3ace0.png)



### :fire: How to test it? (If there is any special consideration or the reviewer does not know how to test it)

### :bookmark_tabs: Others

ifoche commented on .eslintrc.js:
```diff
@@ -13,7 +13,7 @@ module.exports = {
         "no-console": "off",
         "@typescript-eslint/explicit-function-return-type": ["off"],
-        "@typescript-eslint/no-unused-vars": ["error", { argsIgnorePattern: "^_" }],
+        "@typescript-eslint/no-unused-vars": "off",
```
Do we need to switch-off the error on unused vars? What for? In case it's absolutely needed, couldn't we just warn the developer in these situations?

marcgarnica13 replied:
Carried from early stages of development
_/no-unused-vars_ is set as before: ["error", { argsIgnorePattern: "^_" }]

ifoche commented on .eslintrc.js:
```diff
@@ -24,7 +24,7 @@ module.exports = {
         "@typescript-eslint/ban-ts-ignore": "off",
         "@typescript-eslint/no-empty-function": "off",
-        "react-hooks/exhaustive-deps": "error",
+        "react-hooks/exhaustive-deps": "off",
```
Likewise, I guess there's a reason for this, but if you could briefly explain what took you to this decision, it would be great

marcgarnica13 replied:
Carried from early stages of development
_/exhaustive-deps_ is set to "warn". Variables should be included to function array dependencies but adding functions to the dependency array can slow the rendering

ifoche commented on src/components/app/App.tsx:
```diff
@@ -114,7 +114,7 @@ const App = () => {
                 <OldMuiThemeProvider muiTheme={muiThemeLegacy}>
                     <SnackbarProvider>
-                        <HeaderBar appName={"Project Monitoring"} />
+                        <HeaderBar appName={"Google Earth Engine Connector"} />
```
I like the name, I guess we have to ask for an icon to our designer

ifoche replied:
I'll do it

marcgarnica13 commented:
> Thanks @marcgarnica13. I'm requesting here some changes before accepting the PR from a simple end-user perspective, in case they could be easily fixed and implemented:
> 
> 1. I get an error in the localisation part when booting the dev server:
> 
> ```
> > parsing: 34 files
> i18next-scanner: Unable to parse code "{ n: mappingIdsToDelete?.length }"
> i18next-scanner: Error: Line 1: Unexpected token .
>     at ErrorHandler.constructError (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:5012:22)
>     at ErrorHandler.createError (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:5028:27)
>     at Parser.unexpectedTokenError (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:1985:39)
>     at Parser.throwUnexpectedToken (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:1995:21)
>     at Parser.parsePrimaryExpression (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:2366:38)
>     at Parser.inheritCoverGrammar (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:2285:37)
>     at Parser.parseLeftHandSideExpressionAllowCall (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:2899:26)
>     at Parser.inheritCoverGrammar (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:2285:37)
>     at Parser.parseUpdateExpression (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:3004:26)
>     at Parser.parseUnaryExpression (/home/nacho/EyeSeeTea/code/dhis2-web/dev-repos/dhis2-gee-app/node_modules/esprima/dist/esprima.js:3048:26) {
>   index: 25,
>   lineNumber: 1,
>   description: 'Unexpected token .'
> }
> ```
> 
> 1. Could we change the phrasing of the OU and period selections into "Select Organisation Unit" and "Select Period" instead of "edit"?
> 2. Likewise, I would rephrase "Selected mappings" into "Select & map dataset"
> 3. The "Reorder columns", I think you discussed with Arnau and it was better not to rename it as I suggested into "Columns settings", but at least it should work and present the dialog that allows you to select which column to display. Could you make it work?
> 
> Could you have a look and fix those things?
> 
> Thank you so much in advance

1. Fixed: Possible undefined '?' catch was not accepted by i18n parser.
2. Edited.
3. Edited.
4. It seems d2-ui-componet ObjectsTable does not allow the reordering when there is not data found in the table. With data the functinalitiy works smooth.

![image](https://user-images.githubusercontent.com/7198996/77884999-79910800-7266-11ea-95a4-20eed9cca871.png)
